core/src/main/java/io/kestra/core/plugins/DefaultPluginRegistry.java

```diff
@@ -209,33 +198,69 @@ public void register(final RegisteredPlugin plugin) {
-
-        classes.putAll(plugin.getAliases().values().stream().map(e -> {
-                Class<? extends Plugin> pluginClass = (Class<? extends Plugin>) e.getValue();
-                Class<Plugin> pluginBaseClass = plugin.baseClass(pluginClass.getName());
-
-                return new SimpleEntry<>(
-                    ClassTypeIdentifier.create(e.getKey()),
-                    PluginClassAndMetadata.create(plugin, pluginClass, pluginBaseClass, e.getKey())
-                );
-            })
-            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+
+        // Group all classes by identifier (FQN or alias)
+        Map<String, List<Class<? extends Plugin>>> grouped = new HashMap<>();
+        Map<String, String> aliasMap = new HashMap<>();
+
+        // Main plugin classes (FQN)
+        for (Class<?> cls : plugin.allClass()) {
+            String fqn = cls.getName();
+            grouped.computeIfAbsent(fqn, k -> new ArrayList<>()).add((Class<? extends Plugin>) cls);
+        }
+
+        // Aliases: key = alias, value = (alias, class)
+        for (Map.Entry<String, Map.Entry<String, Class<?>>> aliasEntry : plugin.getAliases().entrySet()) {
+            String aliasKey = aliasEntry.getKey();
+            Map.Entry<String, Class<?>> aliasValue = aliasEntry.getValue();
+
+            String identifier = aliasValue.getKey();
+            Class<?> cls = aliasValue.getValue();
+
+            grouped.computeIfAbsent(identifier, k -> new ArrayList<>()).add((Class<? extends Plugin>) cls);
+            aliasMap.put(identifier, aliasKey);
+        }
+
+        // Select the highest priority plugin class per identifier
+        for (Map.Entry<String, List<Class<? extends Plugin>>> entry : grouped.entrySet()) {
+            String identifier = entry.getKey();
+            Class<? extends Plugin> selected = getHighestPriorityPluginClass(entry);
+
+            if (selected == null) {
+                continue;
+            }
+
+            Class<Plugin> pluginBaseClass = plugin.baseClass(selected.getName());
+            String alias = aliasMap.getOrDefault(identifier, null);
+
+            classes.put(
+                ClassTypeIdentifier.create(identifier),
+                PluginClassAndMetadata.create(plugin, selected, pluginBaseClass, alias)
+            );
+        }
+
         return classes;
     }
 
+    private static Class<? extends Plugin> getHighestPriorityPluginClass(Map.Entry<String, List<Class<? extends Plugin>>> entry) {
+        List<Class<? extends Plugin>> classList = entry.getValue();
+
+        Class<? extends Plugin> selected = null;
+        for (Class<? extends Plugin> cls : classList) {
+            io.kestra.core.models.annotations.Plugin annotation = cls.getAnnotation(io.kestra.core.models.annotations.Plugin.class);
+            if (annotation != null && annotation.priority() == io.kestra.core.models.annotations.Plugin.Priority.PRIMARY) {
```

Comment:
Please, create a static method `isPrimary` in the `Plugin` model class for this.

Tips: use `Objects.equals(Priority.PRIMARY, annotation)` instead of checking the `null`